execution-core/src/transfer.rs

```diff
@@ -505,20 +508,224 @@ pub struct PhoenixTransactionEvent {
 }
 
 /// Event data emitted on a moonlight transaction's completion.
+#[cfg_attr(feature = "serde_support", derive(SerdeSerialize, SerdeDeserialize))]
 #[derive(Debug, Clone, Archive, PartialEq, Serialize, Deserialize)]
 #[archive_attr(derive(CheckBytes))]
 pub struct MoonlightTransactionEvent {
     /// The account that initiated the transaction.
+    #[cfg_attr(feature = "serde_support", serde(with = "serde_support::pubk"))]
     pub sender: AccountPublicKey,
     /// The receiver of the funds if any were transferred.
+    #[cfg_attr(
+        feature = "serde_support",
+        serde(with = "serde_support::optional_pubk")
+    )]
     pub receiver: Option<AccountPublicKey>,
     /// Transfer amount
     pub value: u64,
     /// The memo included in the transaction.
+    #[cfg_attr(
+        feature = "serde_support",
+        serde(with = "serde_support::bs58_serde")
+    )]
     pub memo: Vec<u8>,
```

Comment:
Memos should not be bs58 encoded. Prefer using hex for bytes and bs58 encoding only for the bytes of address types like AccountPublicKey